src/ui/run.py

```diff
@@ -3,46 +3,40 @@
 Includes an option for running using livereload.  This helps out a lot
 when developing.
 """
-
-
 import logging
-import os
 
 import waitress
 
 import ui.main as web  # type: ignore
+from config import load_config
 
-
-def serve_debug() -> None:
-    """Workaround for avoiding lamdba. Starts UI in debug."""
-    serve(production=False)
+logger = logging.getLogger(__name__)
+config = load_config()
 
 
-def serve_prod() -> None:
-    """Workaround for avoiding lamdba. Starts UI in production."""
-    serve(production=True)
+def serve_debug() -> int:
+    """Workaround for avoiding lamdba. Starts UI in debug."""
+    ui_server = web.create_app()  # type: ignore
+    ui_server.debug = True
 
+    ui_server.run(use_reloader=False)
+    return 0
 
-def serve(
-    production: bool = True, port: int = 5000, host: str = "0.0.0.0"
-) -> int:
-    """Serve the application."""
-    if production:
-        logger = logging.getLogger("waitress")
-        logger.setLevel(logging.INFO)
-        logger.info("Starting server in production")
 
-        ui_server = web.create_app()
-        ui_server.debug = True
+def serve_prod() -> int:
+    """Workaround for avoiding lamdba. Starts UI in production."""
+    logger = logging.getLogger("waitress")
+    host = config.get("UI", "hostname")
+    port = config.getint("UI", "port")
+    logger.setLevel(logging.INFO)
+    logger.info("Starting server in production")
 
-        waitress.serve(ui_server.wsgi_app, host=host, port=port)  # type: ignore
-    else:
-        ui_server = web.create_app()  # type: ignore
-        ui_server.debug = True
+    ui_server = web.create_app()
+    ui_server.debug = False
 
-        ui_server.run(use_reloader=False)
+    waitress.serve(ui_server.wsgi_app, host=host, port=port)  # type: ignore
     return 0
 
 
 if __name__ == "__main__":
-    exit(serve(False))
+    exit(serve_debug())  # pragma: no cover
```

Comment:
Maybe look at using sys.SystemExit instead.  I believe it to be the "better" approach.

Ref.: https://pythonguides.com/python-exit-command/

Reply:
I think rewriting ui functionality is not part of this merge request. In order to implement the config i had to do some changes to the logic, but kept all refereces functionally the same. In this case `serve(False)` would serve the UI in debug mode, whcih i replicated with the explicit function instead. Feel free to open an issue for this, seeing that UI main is vastly different form core, detection and tracing.